### PR TITLE
Correct rest server

### PR DESCRIPTION
**Pull request by giorgosnty**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes an bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Bitloops Language?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/bitloops/bitloops-language/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the Bitloops Language CLA (if not, email us: contributor@bitloops.com)
- [ ] I created tests which fail with the change
- [ ] Code complies correctly
- [ ] Unit tests pass locally with my changes
- [ ] All existing tests passed
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added/extended necessary documentation
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: transpiler/src/parser/core/grammar/BitloopsParser.g4 [hunk 1]
```diff
@@ -837,15 +827,7 @@ serverDeclaration
...
-    // TODO Remove necessary comma on end of each line
-    : OpenBrace serverInstantiationOption* CloseBrace
-    ;
-
-serverInstantiationOption
-    : serverTypeOption    
-    | serverApiPrefixOption
-    | restServerPort 
-    | customServerOption
+    : OpenBrace (serverTypeOption | serverApiPrefixOption| restServerPort)* CloseBrace
```
Comment (Varagos) [inline transpiler/src/parser/core/grammar/BitloopsParser.g4:830]: Do we need cors option now that customServerOption is gone? (there is no test for, it's used in the todo examples)

**Review by Varagos — APPROVED** [review]

File: transpiler/src/ast/core/BitloopsVisitor/BitloopsVisitor.ts [hunk 2]
```diff
@@ -1014,26 +1017,20 @@ export default class BitloopsVisitor extends BitloopsParserVisitor {
...
+  //   return apiPrefixNode;
+  // }
 
-  //TODO support custom option in the future?
-  // visitCustomServerOption(ctx: BitloopsParser.CustomServerOptionContext): ExpressionNode {
-  //   const res = customServerOptionVisitor(this, ctx);
-  //   return res;
+  // visitRestServerPort(ctx: BitloopsParser.RestServerPortContext): RestServerPortNode {
+  //   return restServerPortVisitor(this, ctx);
   // }
```
Comment (ellik95) [inline transpiler/src/ast/core/BitloopsVisitor/BitloopsVisitor.ts:1034]: comments forgotten?

File: transpiler/src/ast/core/BitloopsVisitor/helpers/evaluation.ts [hunk 3]
```diff
@@ -26,12 +26,6 @@ export const evaluationVisitor = (
   thisVisitor: BitloopsVisitor,
   ctx: BitloopsParser.EvaluationContext,
 ): any => {
-  // const evaluationValues = thisVisitor.visitChildren(ctx)[0];
-  // const evaluation = thisVisitor.evaluationBuilder.withEvaluationValues(evaluationValues).build();
-  // return evaluation;
```
Comment (ellik95) [inline transpiler/src/ast/core/BitloopsVisitor/helpers/evaluation.ts:31]: comments forgotten as well

File: transpiler/src/ast/core/BitloopsVisitor/helpers/setup/corsOptionsVisitor.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+// ): CorsOptionsNode[] => {
+//   //   const metadata = produceMetadata(ctx, thisVisitor);
+
+//   const corsOptions = thisVisitor.visitChildren();
+//   const corsOptionNodes: CorsOptionsNode[] = corsOptions.filter(
+//     (corsOption) => corsOption !== undefined,
+//   );
+
+//   return corsOptionNodes;
+// };
```
Comment (ellik95) [inline transpiler/src/ast/core/BitloopsVisitor/helpers/setup/corsOptionsVisitor.ts:43]: remove comments

File: transpiler/src/ast/core/BitloopsVisitor/helpers/setup/corsOptionsVisitor.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  thisVisitor: BitloopsVisitor,
+  ctx: BitloopsParser.CorsOptionsListStatementContext,
+): CorsOptionsNode => {
+  const metadata = produceMetadata(ctx, thisVisitor);
+
+  const corsOptionsListNode: CorsOptionsListNode[] = thisVisitor.visit(ctx.corsOptionsList());
+
+  const corsOriginNode: CorsOriginNode = corsOptionsListNode.find(
+    (node) => node.getNodeType() === BitloopsTypesMapping.TCorsOrigin,
+  );
```
Comment (ellik95) [inline transpiler/src/ast/core/BitloopsVisitor/helpers/setup/corsOptionsVisitor.ts:22]: Why is this needed? Shouldn't the whole `corsOptionsListNode` be passed to `CorsOptionsNodeBuilder`. If it is needed maybe this logic should be moved to a method inside `CorsOptionsListNode`.

**Comment by Varagos** [comment]
Tests are green, should be merged if ok
